src/vs/workbench/contrib/debug/browser/debugToolBar.ts

```diff
@@ -290,7 +290,12 @@ export class DebugToolBar extends Themable implements IWorkbenchContribution {
 		x ??= this.getStoredXPosition();
 		y ??= this.getStoredYPosition();
 
+		const xMin = 0;
+		// Our max is the entire window minus the width of the toolbar so it doesn't go off screen.
+		// Divide by 2 since the toolbar's zero point is in the middle.
+		const xMax = 1 - this.$el.clientWidth / this.layoutService.activeContainerDimension.width / 2;
```

Comment:
while this is better... it makes me wonder if this needs to factor in the window controls too... can this get moved into the CSS?